trl/trainer/ppo_trainer.py

```diff
@@ -445,30 +445,18 @@ def step(
         if self.is_distributed:
             pad_first = self.tokenizer.padding_side == "left"
 
-            if not self.is_encoder_decoder:
-                model_inputs["input_ids"] = self.accelerator.pad_across_processes(
-                    model_inputs["input_ids"], dim=1, pad_index=self.tokenizer.pad_token_id, pad_first=pad_first
-                )
-                model_inputs["labels"] = self.accelerator.pad_across_processes(
-                    model_inputs["labels"], dim=1, pad_index=-100, pad_first=pad_first
-                )
-                model_inputs["attention_mask"] = self.accelerator.pad_across_processes(
-                    model_inputs["attention_mask"], dim=1, pad_index=0, pad_first=pad_first
-                )
-            else:
+            model_inputs["input_ids"] = self.accelerator.pad_across_processes(
+                model_inputs["input_ids"], dim=1, pad_index=self.tokenizer.pad_token_id, pad_first=pad_first
+            )
+            model_inputs["attention_mask"] = self.accelerator.pad_across_processes(
+                model_inputs["attention_mask"], dim=1, pad_index=0, pad_first=pad_first
+            )
+            if self.is_encoder_decoder:
                 model_inputs["decoder_input_ids"] = self.accelerator.pad_across_processes(
-                    model_inputs["decoder_input_ids"],
```

Comment:
I think this line should have stayed :)

Reply:
yes haha
should be fixed in [3e441da](https://github.com/lvwerra/trl/pull/222/commits/3e441da08340c20b45a547ceb410964928b6690c)